### ISSUE TITLE
UX issue around persistent tooltips

### ISSUE THREAD

**Issue by jareko999** [w3c/wcag#1727]
Hello,

I am a product designer at GitLab and recently I raised an issue in our product around button tooltips covering adjacent buttons and blocking the user from clicking those adjacent buttons. I found [this page](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html) in which it describes: 

"If pointer hover can trigger the additional content, then the pointer can be moved over the additional content without the additional content disappearing". 

I have read through the entire page but do not see clear evidence the value of keeping a persistent 1-3 word tooltip present when hovering this additional content. Why is this in place? 

I see the value of showing a tooltip when hovering the *target* (the link or button) the tooltip describes, but not hovering the actual tooltip itself since it doesn't contain actionable objects or rich content and since it's already being shown on the target's hover/focus. 

If this guideline is to remain, any groups of buttons that contain tooltips that cover adjacent buttons make those adjacent buttons inaccessible unless you "go around" and hover out of the tooltip, then go back to hover the intended button you wanted to click. For mouse users, they also might not be aware of using the keyboard (which is one method outlined in the "Dismissible" section as the acceptable criteria for dismissing a tooltip). Here is our real world example:

![image](https://user-images.githubusercontent.com/13771384/113920454-ccbf6400-97b2-11eb-8396-59c2ca607657.png)

![image](https://user-images.githubusercontent.com/13771384/113920496-d9dc5300-97b2-11eb-81a4-9bb422f41dff.png)

This is a frustrating experience for those who accidentally hover a button, who was then blocked from the button they *intended* to click until they hovered out of the tooltip. Why is it unacceptable to hide the tooltip when hovering it, so it's removed upon exiting the target's container? 

This one specific guideline basically tells me that we have to live with this frustration of accidental hovered buttons with tooltips that block adjacent content. I am interested in getting your thoughts and approach to how we might solve this problem to be more inclusive of mouse users (and those with inaccurate mouse movements) and if keeping a persistent tooltip on it's own hover is beneficial accessibility-wise or if there is a conversation to be had regarding that a tooltip can be dismissible by exiting it's parent target if the "additional content" (as it's described on this page) *does not* contain actionable or rich content.

**Comment by awkawk** [comment]
@jareko999 The core issue is that a screen magnifier user will operate with the screen zoomed in substantially so in some cases the tooltip isn't fully in the visible magnified viewport. In these cases, the user needs to move their mouse onto the tooltip in order to shift the viewport to include the tooltip content. If the tooltip disappears when the user tries to move the mouse onto it, then that content is not accessible to that user.

Check out the tooltip control we have at https://react-spectrum.adobe.com/react-spectrum/Tooltip.html for a sample implementation.

**Comment by jareko999** [comment]
@awkawk Thanks for the quick reply. I understand the core issue now. In regards to a [tooltip delay](https://react-spectrum.adobe.com/react-spectrum/Tooltip.html#warmup--cooldown), are there guidelines to how long a delay should be when showing the tooltip? I think this is something that could solve our problem even with a small delay of 200-300ms.

**Comment by awkawk** [comment]
@jareko999 No, there is no guidance on how much time can pass before the tooltip is shown.

**Comment by jareko999** [comment]
@awkawk Sounds good thanks. I don't know if there's an action item to come out of this, but perhaps [the guidelines](https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus.html) could use a small addition to the `Hoverable` section to recommend a delay when showing a tooltip to prevent:

> Position the additional content so that it does not obscure any other content including the trigger

I think it would be beneficial to those running into the same issue I was. It's not an apparent solution, but a reasonable one to still meet the success criteria. However, that's up to you.

Thanks for your help 👍

**Comment by thierryk** [comment]
> Check out the tooltip control we have at https://react-spectrum.adobe.com/react-spectrum/Tooltip.html for a sample implementation.

Hi @awkawk ,

I'm curious to know why the value of `aria-label` is different than the text inside `<Tooltip>`.

```
<TooltipTrigger>
  <ActionButton aria-label="Edit Name">
    <Edit />
  </ActionButton>
  <Tooltip>Change Name</Tooltip>
</TooltipTrigger>
```

**Comment by thierryk** [comment]
@jareko999 may be the best way to address these issues is to give users the option to make those tooltips "sticky".

**Comment by awkawk** [comment]
@thierryk 
> I'm curious to know why the value for aria-label is different than the text inside <Tooltip>.

The tooltip text is supplementary and doesn't need to mirror the accessibility name value. The HTML spec discusses this related to the title attribute: https://www.w3.org/TR/html52/dom.html#the-title-attribute - "The title attribute represents advisory information for the element, such as would be appropriate for a tooltip."

**Comment by thierryk** [comment]
@awkawk 

Thanks, I asked because looking at the markup I posted the content of the tooltip didn't seem to be exposed to screen-readers but I looked at the source code and noticed the use of `aria-describedby` which makes *a big difference*.

**Comment by patrickhlauke** [comment]
Based on https://github.com/w3c/wcag/issues/1727#issuecomment-815178025 I believe this issue can be closed
